csharp-architectural-patterns/ModularMonolithPatternInCSharp/Inventory/Services/RabbitMqConsumer.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Inventory.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System.Text;
+using System.Text.Json;
+
+namespace Inventory.Services;
+
+internal class RabbitMqConsumer(IServiceScopeFactory scopeFactory, IConfiguration configuration) : IRabbitMqConsumer
+{   
+    private readonly IServiceScopeFactory _scopeFactory = scopeFactory;
+    private readonly IConfiguration _configuration = configuration;
+
+    public void Consume()
+    {
+        // initialize connection
+
+        var connectionFactory = new ConnectionFactory
```

Comment:
I personally think that connections should be long-lived, so doing this here is I think not best practices.

Same for the channels, channels are also meant to be long-lived. Can we make this correct? We also never close the connection/channel.